typing/typemod.ml

```diff
@@ -1829,11 +1829,11 @@ let path_of_module mexp =
    do not contain non-generalized type variable *)
 
 let rec nongen_modtype env = function
-    Mty_ident _ -> false
-  | Mty_alias _ -> false
+    Mty_ident _ -> None
+  | Mty_alias _ -> None
   | Mty_signature sg ->
       let env = Env.add_signature sg env in
-      List.exists (nongen_signature_item env) sg
+      List.find_map (nongen_signature_item env) sg
```

Comment:
I think it might be better to return all non-generalizable variables here. Otherwise, people might think that the single variable that the compiler picked arbitrarily play some sort of key role. We could elide the list if we have more than a handful of those however.

Reply:
I added code to print all the variables. We might need some more tests, especially for modules/classes. It seems not all code paths might be exercised